aim/common/utils.py

```diff
@@ -312,26 +335,27 @@ def inner(*args, **kwargs):
 
 
 def _byteify(data, ignore_dicts=False):
-    if isinstance(data, unicode):
-        return data.encode('utf-8')
+    if isinstance(data, six.text_type):
```

Comment:
The ultimate aim of using `_byteify` is to convert unicode data to bytes (or else it will dump the data into unicode format, since data is in unicode format). 
We are converting into bytes because in Py2, string and bytes are same in Py2. 
While we don't need to convert it into bytes for Py3, because string and unicode are same in Py3.
So no need for conversion in PY3 case.

Reply:
I'm getting confused here. six.text_type means its unicode in both python2 and python3 according to this link,
https://wiki.openstack.org/wiki/Python3#Serialization:_base64.2C_JSON.2C_etc.

And this API is called _byteify() which means it wants to convert unicode string to a byte string right? If so, then we should do,
if isinstance(data, six.text_type):
     return data.encode('utf-8')

which will cover both python2 and python3 case. Make sense? If so, the comment has to be adjusted accordingly also.

Reply:
We are using 'unicode' keyword because unicode is not defined in Py3.
Now, we are using six.text_type because it means unicode for both python2 and python3.

From here on we are differing:

My approach:
(1) Since everything has to be in string text type(this means bytes in Py2 and unicode in Py3) to work everywhere correctly.
(2) To only add specific case for Py3, where we don't want it to be in bytes. (because string is unicode in Py3, and if we convert in bytes).

Your approach:
(1) _byteify() should not change its meaning.

But my question is _byteify() is used in "utils.json_loads()" and this function's aim is to load the json into string format. So if I am assuming correctly then this should handle the Py3 case rather than in _byteify() ?

Reply:
Fixed the change, by following your approach.

Reply:
I think this approach is more clean and less confusing.